unstructured/ingest/v2/cli/base/dest.py

```diff
@@ -0,0 +1,54 @@
+from dataclasses import dataclass
+from typing import Optional, Type
+
+import click
+
+from unstructured.ingest.cli.utils import conform_click_options
+from unstructured.ingest.v2.cli.base.cmd import BaseCmd
+from unstructured.ingest.v2.cli.interfaces import CliConfig
+from unstructured.ingest.v2.logger import logger
+
+
+@dataclass
+class DestCmd(BaseCmd):
+    connection_config: Optional[Type[CliConfig]] = None
+    uploader_config: Optional[Type[CliConfig]] = None
+    upload_stager_config: Optional[Type[CliConfig]] = None
+
+    def cmd(self, ctx: click.Context, **options) -> None:
+        if not ctx.parent:
+            raise click.ClickException("destination command called without a parent")
+        if not ctx.parent.info_name:
+            raise click.ClickException("parent command missing info name")
+        source_cmd = ctx.parent.info_name.replace("-", "_")
+        source_options: dict = ctx.parent.params if ctx.parent else {}
+        conform_click_options(options)
+        try:
+            pipeline = self.get_pipline(
+                src=source_cmd,
+                source_options=source_options,
+                dest=self.cmd_name,
+                destination_options=options,
+            )
+            pipeline.run()
+        except Exception as e:
+            logger.error(f"failed to run destination command {self.cmd_name}: {e}", exc_info=True)
+            raise click.ClickException(str(e)) from e
+
+    def get_cmd(self) -> click.Command:
+        # Dynamically create the command without the use of click decorators
+        fn = self.cmd
+        fn = click.pass_context(fn)
```

Comment:
maybe I'm missing something, why can't we just `fn = click.pass_context(self.cmd)`

Reply:
I wanted to make it intuitive that `click.pass_context()` simply returns a modified version of what was passed in. 
```python
fn = click.pass_context(self.cmd)
```
Doesn't make that as apparent.

Reply:
oh funny, yea, I assumed as much and then was just confused why we were setting it above. I guess dealers choice, don't have strong feelings.